src/core/geocoding/qgsabstractgeocoderlocatorfilter.cpp

```diff
@@ -0,0 +1,80 @@
+/***************************************************************************
+  qgsabstractgeocoderlocatorfilter.cpp
+  ---------------
+  Date                 : August 2020
+  Copyright            : (C) 2020 by Nyall Dawson
+  Email                : nyall dot dawson at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsabstractgeocoderlocatorfilter.h"
+#include "qgsgeocoder.h"
+#include "qgsgeocodercontext.h"
+
+QgsAbstractGeocoderLocatorFilter::QgsAbstractGeocoderLocatorFilter( const QString &name, const QString &displayName, const QString &prefix, QgsGeocoderInterface *geocoder )
+  : mName( name )
+  , mDisplayName( displayName )
+  , mPrefix( prefix )
+  , mGeocoder( geocoder )
+{
```

Comment:
why don't you inherit QgsLocatorFilter here?

Reply:
and have the clone method declared as pure virtual?

Reply:
> why don't you inherit QgsLocatorFilter here?

Ok, done -- I had to add the parent argument to the geocoder filter classes too

> and have the clone method declared as pure virtual?

Reply:
> and have the clone method declared as pure virtual?

It's already pure virtual -- it gets inherited from QgsLocatorFilter as a pure virtual method

Reply:
perfect, thanks